### PR TITLE
Dfci support for using a QemuQ35Pkg virtual system as the device under test

### LINKED ISSUE #83: [Feature]: Support QemuQ35Pkg virtual system for testing the Dfci Test cases
### Feature Overview

Allow a QemuQ35Pkg virtual system to act as the Device Under Test to allow testing and verification of the Dfci unit test end to end test cases.

### Solution Overview

Mostly minor changes - The easiest network capability for QemuQ35 is an E1000 adapter, and the way that is integrated is that ports are forwarded to from the HOST to the Qemu Guest.  Inbound connections are to the HOST IP to a port that is forwarded to the Qemu virtual system.  This means that ping is never forwarded to the Qemu virtual system, and the original Dfci test cases used ping to determine when the Device Under Test was present or not.

### Alternatives Considered

_No response_

### Urgency

Medium

### Are you going to implement the feature request?

I will implement the feature

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mikeytdisco**
## Description

Fixes #83.

This is all changes to the Dfci UnitTest and Docs directories.  There is no change to DFCI functionality.

Quite a number of little changes needed to support Dfci in a QemuQ35Pkg virtual system.   The Q35 environment has fewer settings, so there are fewer settings tested in the Enroll/Unenroll operations.  Due to a pending restructure of the settings test, the current settings test will fail due to settings not in Q35.  All the other tests have been tested.

The restart mechanism needed to be updated to not rely on ping.  Since ping isn't forwarded to Q35, Q35 appeared to never go away.

DFCI_SupportLib.py needed a lot of changes to pass flake8.

This PR will need additional testing using a physical system.

- [ ] Impacts functionality?
  
- [ ] Impacts security?
 
- [x] Breaking change?

This PR will need additional testing using a physical system to ensure the new restart detections mechanism still works.

- [x] Includes tests?
 
- [x] Includes documentation?

## How This Was Tested

Tested with a QemuQ35Pkg virtual system.

## Integration Instructions

N/A

### CODE REVIEW

File: DfciPkg/Docs/QemuQ35Pkg/ReadMe.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
+# Building and Running QemuQ35Pkg with DFCI
+
+QemuQ35Pkg virtual machine environment is available at
```
Comment (apop5) [inline DfciPkg/Docs/QemuQ35Pkg/ReadMe.md:3]:
```suggestion
QemuQ35Pkg virtual machine environment is available
```

File: DfciPkg/Docs/QemuQ35Pkg/ReadMe.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+This document describes how to build and Run a QemuQ35Pkg virtual device to
+exercise the DFCI test cases.
+
+## Getting started
+
+You will need the following:
+
+1. Knowledge that the sub directory DfciDutFiles is used for both the build and run scripts for
+Q35, the emulated USB drives used to install windows, and setup the DFCI Test solution into the QemuQ35Pkg VM.
+2. An install image for Windows 11 (available when you purchase a license for Windows 11).
```
Comment (apop5) [inline DfciPkg/Docs/QemuQ35Pkg/ReadMe.md:15]: I believe it is possible to download a windows 11 iso without a valid license. Is the license required to enable DFCI support?

File: DfciPkg/Docs/QemuQ35Pkg/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    ```text
+        RunDfci -f
+    ```
+
+You will be presented with a Microsoft Surface like firmware interface
+
+![FrontPage](ScreenShots/FrontPage.png)
+
+The QemuQ35Pkg virtual system is used by Project Mu for various testing activities, and had
+setting set a certain way for that test environment.
```
Comment (apop5) [inline DfciPkg/Docs/QemuQ35Pkg/ReadMe.md:113]:
```suggestion
settings set a certain way for that test environment.
```

**Review by apop5 — APPROVED** [review]

**Review by Flickdm — APPROVED** [review]

File: DfciPkg/Docs/QemuQ35Pkg/ReadMe.md (hunk 2 above)
Comment (mikeytdisco) [reply DfciPkg/Docs/QemuQ35Pkg/ReadMe.md:15]: Just following the party line.  I don't think I should be the one publishing workarounds, even though there are some that are published - and the work arounds have to change as release of Windows change.  As for DFCI, there is no special license for use.

**Pull request merged**